contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -131,47 +135,75 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 	}), middleware.After)
 }
 
-func resourceNameFromParams(requestInput middleware.InitializeInput, awsService string) (string, string, bool) {
-	var k, v string
+func resourceTagsFromParams(requestInput middleware.InitializeInput, awsService string, region string) (map[string]string, bool) {
+	tags := make(map[string]string)
```

Comment:
Nit: this is going to be an allocation each time the function returns. It's a good candidate to use a `sync.Pool` or similar to reuse maps. Additionally, I'd suggest to allocate it with a hint of `2`.